### ISSUE TITLE
haxe.Int64 typing error

### ISSUE THREAD

**Issue by ncannasse** [HaxeFoundation/haxe#10703]
The following fails to compile with `haxe.Int64 should be Int` which is a bit weird:

```haxe
	static function foo(v:Int,b:Bool) : haxe.Int64 {
		return v << (b ? 2 : 4);
	}
```

**Comment by Simn** [comment]
Yes our expected type handling with binops is confusing sometimes. I think we push the expected type into both operands, but then `<<` calls `unify_int` on them.
